src/libra_py/dynamics/qtag/qtag_prop.py

```diff
@@ -60,7 +60,7 @@ def propagate(dyn_params, qpas, coeff, surf_pops):
     dt = params["dt"]
     decpl = params["decpl_den"] 
     iM = params["iM"]  # MATRIX(ndof, 1)
-
+    states = params["states"]
```

Comment:
`states` can be the input parameter here, but not in the C++ side - by that time, we should have called the appropriate initialization function, create the dynamical variables with the states of all trajectories or such and pass them directly to the propagation routine